web/packages/shared/utils/highbar.ts

```diff
@@ -68,33 +68,51 @@ export function arrayObjectIsEqual(arr1: unknown[], arr2: unknown[]): boolean {
   );
 }
 
-/** Recursively compares two values. */
-export function equalsDeep(val1: unknown, val2: unknown) {
-  if (!isObject(val1)) {
+/**
+ * Recursively compares two values. If `ignoreUndefined` is set to `true`,
+ * treats fields set to `undefined` as equal to fields that don't exist at all.
+ */
+export function equalsDeep(
+  val1: unknown,
+  val2: unknown,
+  ignoreUndefined?: boolean
```

Comment:
Would be cool to make the 3rd argument an object:
```suggestion 
  options?: {
    /**
     * If `true`, treats fields set to `undefined`
     * as equal to fields that don't exist at all.
     */
    ignoreUndefined?: boolean;
  }
```

Reply:
Yeah, I thought about it, but was a bit on the fence about it. But now you mentioned it, I'll change it. :)